src/microbe_stage/Microbe.Contact.cs

```diff
@@ -548,20 +553,24 @@ public void Kill()
             var sceneToUse = new ChunkConfiguration.ChunkScene();
 
             // Try all organelles in random order and use the first one with a scene for model
-            foreach (var organelle in organelles.OrderBy(_ => random.Next()))
+            foreach (var organelle in organellesAvailable.OrderBy(_ => random.Next()))
             {
                 if (!string.IsNullOrEmpty(organelle.Definition.CorpseChunkScene))
                 {
                     sceneToUse.LoadedScene = organelle.Definition.LoadedCorpseChunkScene;
+                    organellesAvailable.Remove(organelle);
                     break;
                 }
 
                 if (!string.IsNullOrEmpty(organelle.Definition.DisplayScene))
                 {
                     sceneToUse.LoadedScene = organelle.Definition.LoadedScene;
                     sceneToUse.SceneModelPath = organelle.Definition.DisplaySceneModelPath;
+                    organellesAvailable.Remove(organelle);
                     break;
                 }
+
+                organellesAvailable.RemoveAll(item => item == organelle);
```

Comment:
Emm ... I guess I'll just take back what I said before. We'd better delete this line, because if this line is executed you'll certainly get a ModifiedException the next loop. If we really want to delete without exceptions, we may need more time to do that.